### ISSUE TITLE
Component-level imports for module-related components

### ISSUE THREAD

**Issue by TheTomasJ** [angular/angular#58567]
### Which @angular/* package(s) are relevant/related to the feature request?

core

### Description

Standalone components are great. But sometimes you still need to use modules (DDD, bound components, etc..). Large modules then have lot of imports and potentially can include unused code.

### Proposed solution

I would like to benefit from combination of both: Component can manage its own dependencies but must be imported within a parent module. 
With such possibility, domain modules will reduce its complexity and risk of unused imports whereas it still can cover modularity principle where needed.

option 1: allow imports on component level also for non-standalone components
option 2: introduce mechanism to prevent direct imports of standalone component (protected standalone component)

### Alternatives considered

Module which is reexporting standalone components, but those components are still available for import without the parent module.

**Comment by JeanMeche** [comment]
Hi, 
This feels like something that is related to architecture boundaries (and not technical like what modules actually are). 

Have you looked at solutions likes [NX's module boundary](https://nx.dev/features/enforce-module-boundaries) or [`eslint-plugin-boundaries`](https://www.npmjs.com/package/eslint-plugin-boundaries) ?

**Comment by TheTomasJ** [comment]
You are right, the proposed option 2 is architectural boundary. So "DDD" use case is not good example.

But my original idea was to "clean-up" the modules, by declaring most of dependencies where are neccessary. 

The fact is, that this is not critical problem in small functional-module and DDD modules should be handled in different way. So maybe some official response to this is enough (since I talked about this with some ng-devs for a months).

**Comment by alxhub** [comment]
You're correct that standalone combines two concepts:

* A component declares its own dependencies
* A component can be used without importing an NgModule

Since our goal with standalone was to make NgModules unnecessary, it made sense to combine these two concepts into a single feature. Having this property allows us to build features like `@defer`, `Route.loadComponent`, etc which _rely_ on components not requiring an NgModule to load.

The Angular team does not see NgModules as a valuable or recommended architectural pattern, and we recommend that all new Angular code is written as standalone.
